bundles/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/handler/PLCBridgeHandler.java

```diff
@@ -144,62 +160,70 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             return;
         }
 
-        PLCLogoClient localClient = client;
-        String channelId = channelUID.getId();
-        Channel channel = thing.getChannel(channelId);
-        Layout layout = LOGO_CHANNELS.get(channelId);
-        if ((localClient != null) && (channel != null) && (layout != null)) {
-            byte[] buffer = new byte[layout.length];
+        final var client = this.client;
+        final var channelId = channelUID.getId();
+        final var layout = LOGO_CHANNELS.get(channelId);
+        if ((client != null) && (layout != null)) {
+            var buffer = new byte[layout.length()];
             Arrays.fill(buffer, (byte) 0);
-            int result = localClient.readDBArea(1, layout.address, buffer.length, S7Client.S7WLByte, buffer);
+            int result = client.readBytes(layout.address(), buffer.length, buffer);
             if (result == 0) {
-                if (RTC_CHANNEL.equals(channelId)) {
-                    ZonedDateTime clock = ZonedDateTime.now();
-                    if (!LOGO_0BA7.equalsIgnoreCase(getLogoFamily())) {
-                        try {
-                            int year = clock.getYear() / 100;
-                            clock = clock.withYear(100 * year + buffer[0]);
-                            clock = clock.withMonth(buffer[1]);
-                            clock = clock.withDayOfMonth(buffer[2]);
-                            clock = clock.withHour(buffer[3]);
-                            clock = clock.withMinute(buffer[4]);
-                            clock = clock.withSecond(buffer[5]);
-                        } catch (DateTimeException exception) {
-                            clock = ZonedDateTime.now();
-                            logger.info("Return local server time: {}.", exception.getMessage());
+                switch (channelId) {
+                    case RTC_CHANNEL -> {
+                        rtc = ZonedDateTime.now();
+                        if (!LOGO_0BA7.equalsIgnoreCase(getLogoFamily())) {
+                            try {
+                                final var year = rtc.getYear() - rtc.getYear() % 100;
+                                final var date = LocalDate.of(year + buffer[0], buffer[1], buffer[2]);
+                                final var time = LocalTime.of(buffer[3], buffer[4], buffer[5]);
+                                rtc = ZonedDateTime.of(date, time, ZoneId.systemDefault());
```

Comment:
Sure you want the server timezone, not the configured timezone of openHAB? @jlaur is a big fan of using Instant, so maybe that would even be better.

Reply:
Strickly speaking, it's about time zone set in PLC. From my point of view, the best possible approximation for is the system default. At least till i found the way to get it from PLC...